### PR TITLE
Optimizations to the tiling calculator

### PR DESCRIPTION
**Pull request by micheles**
See https://bugs.launchpad.net/oq-engine/+bug/1465021.
The tests are running here: https://ci.openquake.org/job/zdevel_oq-risklib/727

### CODE REVIEW

File: openquake/commonlib/source.py [hunk 1]
```diff
@@ -688,33 +685,106 @@ def filter_and_split(src, sourceprocessor):
...
+        """
+        self.infos.append(
+            (info.trt_model_id, info.source_id, info.source_class,
+             len(info.sources), info.filter_time, info.split_time))
+        return acc + {info.trt_model_id: info.sources}
+
+    def process(self, csm):
+        """
+        :param csm: a CompositeSourceModel instance
+        :param monitor: a monitor object
```
Comment (ptormene) [inline openquake/commonlib/source.py]: monitor is not in the signature

File: openquake/commonlib/source.py [hunk 2]
```diff
@@ -688,33 +685,106 @@ def filter_and_split(src, sourceprocessor):
...
+
+class SourceFilterSplitter(SourceFilter):
+    """
+    Filter and split in parallel the sources of the given CompositeSourceModel
+    instance. An array `.source_info` is added to the instance, containing
+    information about the processing times and the splitting process.
 
+    :param sitecol: a SiteCollection instance
+    :param maxdist: maximum distance for the filtering
+    :param asd: area source discretization
```
Comment (ptormene) [inline openquake/commonlib/source.py]: wrong name for asd

Comment (micheles) [inline openquake/commonlib/source.py]: This is actually fixed in the other pull request

File: openquake/commonlib/source.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/source.py]: Fixed

File: openquake/commonlib/source.py [hunk 3]
```diff
@@ -688,37 +685,108 @@ def filter_and_split(src, sourceprocessor):
...
+        return seqtime, partime
+
+    def update(self, csm, sources_by_trt):
+        """
+        Store the `source_info` array in the composite source model.
+
+        :param csm: a CompositeSourceModel instance
+        :param sources_by_trt: a dictionary trt_model_id -> sources
+        """
+        self.infos.sort(key=lambda o: o[4] + o[5], reverse=True)
```
Comment (ptormene) [inline openquake/commonlib/source.py]: Perhaps you might add a comment to describe the sorting key

**Comment by ptormene** [comment]
Besides the minor comment, LGTM

File: openquake/commonlib/source.py (hunk 3 above)
Comment (micheles) [inline openquake/commonlib/source.py]: Yes, I am doing this in the other pull request

**Pull request merged**